airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -158,6 +158,8 @@ class KubernetesPodOperator(BaseOperator):
         suffix if random_name_suffix is True) to generate a pod id (DNS-1123 subdomain,
         containing only [a-z0-9.-]).
     :param random_name_suffix: if True, will generate a random suffix.
+    :param job_id_as_suffix: if True, will add ti.job_id to the pod name. random_name_suffix, if set to True
+        takes precedence over this parameter.
```

Comment:
This will be very confusing.
If we want this need we need to do something similar to https://github.com/apache/airflow/pull/30718
e.g deprecate `random_name_suffix` and add `name_suffix` with options of `(None, "random", "job_id")`

On a more high level note I don't think this will work as you expect?
as noted in random_name_suffix the chars allowed are `[a-z0-9.-]` so what will happen when task_id contains illegal chars? I suspect that your goal is to match the task_id of airflow to the pod name for easier find but when it's not exact match and you start playing around with modifying the names then it defeat the purpose

WDYT?

Reply:
Hi, I agree with your proposal to change the suffixing logic instead. I don't know how the deprecation process would work for this, though.

This is using `ti.job_id` which I think is assigned by airflow, `task_id` is separate and can be supplied in `KubernetesPodOperator`. We shouldn't use that for the reason you state. I have not seen a way to modify the `job_id`.

Reply:
I've added another proposal which makes the following changes
- Use name_suffix instead of two arguments to control the suffix operation as suggested
- Make hostname suffix modification optional via `add_suffix_to_hostname`. I am not aware of situations where hostname could/should be different from pod name but the random suffix has not been applied to hostname before.
- Change default of `random_suffix` to False and notify about deprecation if it's set to True.
- Clean `_create_pod_id` as some options were never given. Set the full pod name with suffix inside to take account both random and job_id based suffix.

Reply:
Consulted with @jedcunningham also on this one.

Can you clarify why `pod_mutation_hook` and/or making the name parameter `templated_field` is not sufficient ?

Reply:
Thank you both for the feedback and the suggestions. I think the pod_mutation_field could work but need to make sure the pod name stays within the length limit. The `templated_field` option sounds fine too but requires a change in the op to skip the name validity checking in the constructor. Do you think this is worth pursuing, I could prepare a PR for that?

I found `pre_execute` to be most suitable for my purposes after all. Since the context of the job to be launched is available I can override the name before `execute` is called. I will close the PR for now.